src/validators.ts

```diff
@@ -190,3 +191,15 @@ export function either<F, S>(
     return result;
   };
 }
+
+export function decodeBase64<T>(validator: Validator<T>): Buffer {
```

Comment:
I'm a bit lost on this validator, @Mossop do you have any suggestions? Thank you.

Reply:
You probably want something like:
```typescript
export function base64(val: any): Buffer {
  if (typeof val != "string") {
    throw new Error(`Expected a base64 encoded string but received ${repr(val)}`);
  }

  try {
    return Buffer.from(val, "base64");
  } catch (e) {
    throw new Error(`Expected a base64 encoded string but received ${repr(val)}`);
  }
}
```

Reply:
Thank you, works perfectly! :+1: